src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceSync.kt

```diff
@@ -3,17 +3,19 @@ package com.cognifide.gradle.aem.instance
 import com.cognifide.gradle.aem.common.*
 import com.cognifide.gradle.aem.common.http.RequestException
 import com.cognifide.gradle.aem.common.http.ResponseException
+import com.cognifide.gradle.aem.instance.content.ContentManipulator
+import com.cognifide.gradle.aem.instance.content.Node
 import com.cognifide.gradle.aem.pkg.*
 import com.cognifide.gradle.aem.pkg.tasks.PackageCompose
-import java.io.File
-import java.io.FileNotFoundException
 import org.apache.commons.io.FilenameUtils
 import org.jsoup.Jsoup
 import org.jsoup.parser.Parser
 import org.zeroturnaround.zip.ZipUtil
+import java.io.File
+import java.io.FileNotFoundException
 
 @Suppress("LargeClass", "TooManyFunctions")
-class InstanceSync(aem: AemExtension, instance: Instance) : InstanceHttpClient(aem, instance) {
+class InstanceSync(aem: AemExtension, instance: Instance) : InstanceHttpClient(aem, instance), ContentManipulator {
```

Comment:
@pun-ky should InstanceSync be extended with those features? How the DSL will look like?
@szymon-owczarzak can you update README.md and post examples of how this gonna be used?

Reply:
I suggested adding these methods directly to sync, eventually we could think about nesting these methods under subobject, yep composition over extending to big object

how about 

```kotlin

class InstanceSync {
 
val repository = InstanceRepository(this)

fun repository (callback: InstanceRepository.() -> Unit) {
    repository.apply(callback)
}

}

```

then

```kotlin
aem.anyInstance.sync {
    repository.getNode("/content/example/demo")
}
```

to sum up,

`ContentManipulator` -> `InstanceRepository`

all methods not put directly inside `InstanceSync` but organized in separate `InstanceRepository` object

Reply:
After catchup with @mierzwid we come up with Nodes class name so we will have:

```
aem.anyInstance.sync.nodes {
    get("/content/example/demo")
}
```